plugins/experimental/ts_lua/ts_lua.c

```diff
@@ -109,29 +110,49 @@ TSRemapNewInstance(int argc, char *argv[], void **ih, char *errbuf, int errbuf_s
     return TS_ERROR;
   }
 
-  ts_lua_instance_conf *conf = TSmalloc(sizeof(ts_lua_instance_conf));
-  if (!conf) {
-    strncpy(errbuf, "[TSRemapNewInstance] TSmalloc failed!!", errbuf_size - 1);
-    errbuf[errbuf_size - 1] = '\0';
-    return TS_ERROR;
+  ts_lua_instance_conf *conf = NULL;
+
+  // check to make sure it is a lua file and there is no parameter for the lua file
+  if (fn && (argc - optind < 2)) {
+    TSDebug(TS_LUA_DEBUG_TAG, "[%s] checking if script has been registered", __FUNCTION__);
+    char script[TS_LUA_MAX_SCRIPT_FNAME_LENGTH];
```

Comment:
`char script[TS_LUA_MAX_SCRIPT_FNAME_LENGTH+1];`, to account for a terminating nul?

Reply:
no need. we checked the length of the string above before reaching here.